lib/SILOptimizer/Mandatory/TFDeabstraction.cpp

```diff
@@ -2522,13 +2524,41 @@ void TFDeabstractionPass::run() {
                                    fn.getName().str().c_str());
 
     TFDeabstraction(fn, tfc, constantEvaluator, PM).doIt();
+    partitionedFunctions.insert(&fn);
 
     // TODO(clattner): This should eventually be the driver that kicks off
     // the partitioning pass as part of it, and the partitioning and later
     // passes are just function passes that are invoked by this one.  Until
     // we are ready for that, let them run later in the pipeline after the
     // other optimization and cleanup passes.
   }
+
+  // Deabstract stragglers that were left out in the previous iteration. These
+  // are functions that are *still* referred to in the code and operate on
+  // tensor values, but have not been partitioned. It can happen in the
+  // following case, for instance, where `foo` is an external function that
+  // takes a for which we do not have the body:
+  //   main() {
+  //     foo() { $0 -= 0.1 * $1 }
+  //   }
+  //
+  // In the common case where we have the body of foo(), it gets inlined, and
+  // therefore, the closure gets inlined too. However, if the body of foo() is
+  // not available, the closure never gets inlined. To ensure that the call to
+  // the closure within foo() works correctly, we will have to deabstract and
+  // partition the closure.
+  //
+  // (Note the body of a function may be missing when we are linking against a
+  // library or in the REPL context where the function was defined on a
+  // different line.)
```

Comment:
nit: "different line" might not be the right description.

@marcrasi showed me earlier that you can copy/paste multi-line text into REPL, and that'll be executed as one "cell".

Reply:
Calling it REPL line now. Not sure if that is the right terminology.